apps/system/js/app_chrome.js

```diff
@@ -340,6 +373,67 @@
     window.dispatchEvent(new CustomEvent('global-search-request'));
   };
 
+  AppChrome.prototype.onClickSiteIcon = function onClickSiteIcon() {
+    this.pinningPref && this.pinningPref.get(function(enabled) {
```

Comment:
Is it possible to listen to setting change and add/remove the event listener rather than check the setting every time you press the icon?